lib/struct/logger.ts

```diff
@@ -44,20 +58,45 @@ export class Logger {
     return [channel, guild];
   }
 
-  private genLogMessage(
+  private async genLogMessage(
     user: User,
     domain: string,
     taken: string[],
+    channel: string,
+    guildid: string,
     failed: string[]
   ) {
     if (taken.length) {
-      return `:hammer: Phishing URL sent by ${user} (**${user.tag}**, \`${
-        user.id
-      }\`). Actions: ${taken.map(a => `\`${a}\``).join(', ')} ${
+      const defaultMessage = `:hammer: Phishing URL sent by ${user} (**${
+        user.tag
+      }**, \`${user.id}\`). Actions: ${taken.map(a => `\`${a}\``).join(', ')} ${
         failed.length
           ? ':warning: Failed: ' + failed.map(a => `\`${a}\``).join(', ')
           : ''
       }\n> \`${domain}\``;
+      let message: string = defaultMessage;
+      const guildConfig = await this.client.db.guildConfigs.get(guildid);
+      if (
+        guildConfig &&
+        guildConfig.logFormat &&
+        guildConfig.logFormat.length > 0
+      ) {
+        const logFormat = guildConfig.logFormat;
+        const actions = `${taken.map(a => `\`${a}\``).join(', ')} ${
+          failed.length
+            ? ':warning: Failed: ' + failed.map(a => `\`${a}\``).join(', ')
+            : ''
+        }`;
+        message = logFormat
+          .replace(/{actions}/g, actions)
+          .replace(/{domain}/g, domain)
+          .replace(/{offender}/g, `${user} (**${user.tag}**, \`${user.id}\`)`)
+          .replace(/{offenderTag}/g, user.tag)
+          .replace(/{offenderId}/g, user.id.toString())
```

Comment:
IMO these would be better as `offender.id` and `offender.tag`.

Additionally, please add `{offender.username}` and `{offender.discriminator}|`

Reply:
Will do